### PR TITLE
implement ServerStatusDaemon

### LINKED ISSUE #2575: improve memory management documentation
**Feature Description**
Need to complete the memory management documentation

### LINKED ISSUE #2658: implement StatusServerDaemon
**Feature Description**
In support of multiuser Arkouda, create an optional StatusServiceDaemon with a separate ZMQ endpoint that will eventually deliver the following features:

1. readiness/liveness probe (move current ruok endpoint to the StatusServiceDaemon)
2. per-locale memory status (avail/allocated to chapel/used within each locale Chapel process)
3. currently registered pdarrays with corresponding size and type for each
4. number and user names of currently-connected users
5. idleness probe (is server currently executing a request)
6. request type/requesting user for currently executing server request, if applicable

The initial implementation will focus solely on delivering a readiness/liveness probe which will enable users to confirm if an Arkouda server is up. 

**Additional context**
Note: this feature is not currently available due to the blocking nature of the Request/Reply ZMQ socket for the cmd request and metrics endpoints. Stated differently, it is possible that readiness/liveness probe calls would be blocked by another user request if implemented on the cmd request endpoint.

In addition, it is understood that the availability of a vCPU on the locale0 host to respond to the any StatusServerDaemon is variable due the Chapel threading model, so there may be times that StatusServerDaemon requests will be slow to respond.

### PR DESCRIPTION
**Pull request by hokiegeek2**
Closes #2658 and closes #2575 

This PR does the following:

1. implements a liveness/readiness probe in the form of the ServerStatusDaemon
2. adds OS checking for MemoryMgmt module which only works on Linux and Unix systems
3. adds/updates README content and Chapel doc comments

### CODE REVIEW

**Review by jaketrookman — APPROVED** [review]
Looks good to me

**Review by TheOddCzar789 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -103,11 +103,10 @@ This yielded a >20TB dataframe in Arkouda.
    - [Connecting to Arkouda](#run-ak-connect)
 5. [Logging](#log-ak)
 6. [Type Checking in Arkouda](#typecheck-ak)
-7. [Environment Variables](#env-vars-ak)
-8. [Versioning](#versioning-ak)
-9. [External Systems Integration](#external-integration)
-10. [Metrics](#metrics)
-11. [Contributing](#contrib-ak)
+7. [Versioning](#versioning-ak)
```
Comment (Ethan-DeBandi99) [inline README.md:106]: I just want to check that we don't need the Environment Variables section any longer? It doesn't look like it, but want to make sure.

**Review by Ethan-DeBandi99 — APPROVED** [review]
Looks good. Single quick verification (more of a sanity check).

File: README.md [hunk 2]
```diff
@@ -249,22 +248,46 @@ is saved to the .arkouda/tokens.txt file for re-use.
...
 ```
 
-The max memory utilized by each locale can be set in one of two ways: percentage of physical memory or a limit set in bytes. By default, the max per-locale memory is set to ninety (90) percent of the physical memory on each Arkouda locale host. If another percentage is desired, this is set via the --perLocaleMemLimit startup parameter. For example, to set max memory utilized by each locale to seventy (70) percent of physical memory, the Arkouda startup command would be as follows:
+The Arkouda dynamic memory limit approach sets the per-locale memory limit based upon a configurable percentage of available memory on each locale host. Prior to the execution of each comand, the MemoryMgmt [localeMemAvailable](https://github.com/Bears-R-Us/arkouda/blob/e4a48c52eb00097e6e1dfa365cbc586e2e988a85/src/MemoryMgmt.chpl#L133) function does the following on each locale:
+
+1. Verifies if the projected, additional per-locale memory required by the incoming command exceeds the memory currently allocated to Arkouda. If the projected, additional memory is within the memory currently allocated to Arkouda on each locale, the command is allowed to proceed.
```
Comment (pierce314159) [inline README.md:261]:
```suggestion
1. Verifies the projected, additional per-locale memory required by the incoming command does not exceed the memory currently allocated to Arkouda. If the projected, additional memory is within the memory currently allocated to Arkouda on each locale, the command is allowed to proceed.
```

**Review by pierce314159 — APPROVED** [review]
Looks good! Few super small comments

File: README.md [hunk 3]
```diff
@@ -378,17 +401,13 @@ type checking require type hints. Consequently, to opt-out of type checking, sim
...
-The various Arkouda aspects (compilation, run-time, client, tests, etc.) can be configured using a number of environment
-variables (env vars).  See the [ENVIRONMENT](ENVIRONMENT.md) documentation for more details.
 
 <a id="versioning-ak"></a>
 ## Versioning <sup><sup><sub><a href="#toc">toc</a></sub></sup></sup>
 Beginning after tag `v2019.12.10` versioning is now performed using [Versioneer](https://github.com/python-versioneer/python-versioneer)
 which determines the version based on the location in `git`.
 
-An example using a hypothetical tag `1.2.3.4`
+An example using a hypothetical tag 1.2.3.4
```
[the remark this answers is not in the document]
Comment (hokiegeek2) [reply README.md:410]: @pierce314159 it appears to jack up the formatting of the README fille

File: README.md (hunk 1 above)
Comment (hokiegeek2) [reply README.md:106]: @Ethan-DeBandi99 Yup, gotta be sure, and the ENVIRONMENT.md file has been gone for awhile

File: README.md [hunk 4]
```diff
@@ -249,22 +248,46 @@ is saved to the .arkouda/tokens.txt file for re-use.
...
 
 ```
-export CHPL_RT_NUM_THREADS_PER_LOCALE=16
+./arkouda_server --memMax=100000000000
 ```
 
-The max memory utilized by each locale can be set in one of two ways: percentage of physical memory or a limit set in bytes. By default, the max per-locale memory is set to ninety (90) percent of the physical memory on each Arkouda locale host. If another percentage is desired, this is set via the --perLocaleMemLimit startup parameter. For example, to set max memory utilized by each locale to seventy (70) percent of physical memory, the Arkouda startup command would be as follows:
+The Arkouda dynamic memory limit approach sets the per-locale memory limit based upon a configurable percentage of available memory on each locale host. Prior to the execution of each comand, the MemoryMgmt [localeMemAvailable](https://github.com/Bears-R-Us/arkouda/blob/e4a48c52eb00097e6e1dfa365cbc586e2e988a85/src/MemoryMgmt.chpl#L133) function does the following on each locale:
```
[the remark this answers is not in the document]
Comment (hokiegeek2) [reply README.md]: @pierce314159 good catch, fixed!

**Review by pierce314159 — APPROVED** [review]

**Pull request merged**